packages/core/state/selection/selectors.ts

```diff
@@ -76,3 +77,37 @@ export const getUnavailableAnnotationsForHierarchy = createSelector(
             allAnnotations.filter((annotation) => !availableAnnotations.includes(annotation.name))
         )
 );
+
+// COMPOSED SELECTORS
+export const getSortedAnnotations = createSelector(
+    [getAnnotations, getRecentAnnotations],
+    (annotations: Annotation[], recentAnnotationNames: string[]) => {
+        // Create Array of annotations from recentAnnotationNames
+        const recentAnnotations = recentAnnotationNames.flatMap((name) =>
+            annotations.filter((annotation) => annotation.name === name)
+        );
+
+        // get the File name annotaion in a list (if Present)
+        const fileNameAnnotation = annotations.filter(
+            (annotation) =>
+                annotation.name === AnnotationName.FILE_NAME || annotation.name === "File Name"
+        );
+
+        // combine all annotation lists
+        const combinedAnnotations = fileNameAnnotation.concat(
```

Comment:
Making sure I understand -this results in:  FileName, (recent annotations if any, in original order), (all other annotations, sorted)